### PR TITLE
Re-Form - Bare Metal Forms and Helpers

### PR DESCRIPTION
**Pull request by ad9311**
In this PR I present Re-Form

This app has users controller and user model.</br>
- The root page is /users#new
- The app is capable of creating new users and edit them
- Users' username and email must be unique. Passwords and username must be between 6 - 16 characters.
- new.html.erb has commented-out lines of code as per requested by the Odin project.

### CODE REVIEW

**Review by mke2111 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🔴 ♻️ 

## Hey Happy Coder, 👋🏽 
**This is a great project you have going on, well-done lad**. 

### Well-done 💯 
    1. The readme is clear and concise.
    2. The app runs just fine.

**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

### REQUIRED

- [ ] Add navigation buttons to all your pages. I had to input the edit route to test it out.
- [ ] Check inline comments for sections to improve.

#

### Kindly make all the requested changes and submit another CRR. 💯 

**Cheers 🍹 and Happy coding 💻** 

<hr>

Mention @mke2111 in this thread and I'll be glad to help. 😄 
### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class UsersController < ApplicationController
+  # before_action :set_user, only: %i[ show edit update destroy ]
```
Comment (mke2111) [inline app/controllers/users_controller.rb:2]: - [ ] Remove commented-out code.

File: app/views/users/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<h1>Edit User</h1>
+<%= form_with(model: @user) do |form| %>
```
Comment (mke2111) [inline app/views/users/edit.html.erb:2]: - [ ] The edit html file should use the `form_for` tag. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_edit.png)

**Comment by ad9311** [comment]
I apply the following changes:
- Remove unnecessary commented-out code.
- Replace form_with to form_for for edit.html.erb
- Add links for editing a user. The links are in the new actions.

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :green_circle: 

## Hello @ad9311, thank you for making all the required changes. Your project looks complete and good luck with your next ones :+1: 

# HAPPY CODING :computer: :desktop_computer:

**Pull request merged**